code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -43,9 +54,13 @@
 		to_chat(usr, "<span class='warning'>[src] will not function on a deceased cyborg.</span>")
 		return
 	if(module_type && !istype(R.module, module_type))
-		to_chat(R, "<span class='warning'>Upgrade mounting error!  No suitable hardpoint detected!</span>")
+		to_chat(R, "<span class='warning'>Upgrade mounting error! No suitable hardpoint detected!</span>")
 		to_chat(usr, "<span class='warning'>There's no mounting point for the module!</span>")
 		return
+	var/obj/item/borg/upgrade/u = locate(type) in R
+	if(u && !allow_duplicate)
+		to_chat(usr, "<span class='notice'>This unit already has [src] installed!</span>")
```

Comment:
I'm pretty sure that `mob/user` is preferred over `usr`.

Reply:
Less preferred and more required